### PR TITLE
#32-InstallmentPage

### PR DESCRIPTION
**Pull request by AsmaMohame**

### CODE REVIEW

File: retail-managment-system/rms-ui/src/app/domain/installment/models/installment.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+export interface Installment {
+     id: number;
+    version: number;
+    installmentAmount :number;
+    paymentAmount : number;
+    dueDate : string;
+    paymentDate : string;
+    status : number ; 
+    orderDTO : Order
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/app/domain/installment/models/installment.ts:12]: orderDTO is not the same property in swagger.

File: retail-managment-system/rms-ui/src/app/pages/installments/installment/installment.component.html [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+      </td>
+    </ng-container>
+    <tr mat-header-row *matHeaderRowDef="displayedColumns"></tr>
+    <tr mat-row *matRowDef="let row; columns: displayedColumns"></tr>
+    </table>
+  </div>
+  
+  
+    <div class="content col-lg-4 col-md-4 col-sm-6 bg-light">
+      <section>
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/app/pages/installments/installment/installment.component.html:75]: are you need this tag?

File: retail-managment-system/rms-ui/src/app/pages/installments/installment/installment.component.ts [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+import { InstallmentRepositry } from 'src/app/domain/installment/installment.repositry';
+import { Installment } from 'src/app/domain/installment/models/installment';
+
+@Component({
+  selector: 'app-installment',
+  templateUrl: './installment.component.html',
+  styleUrls: ['./installment.component.scss']
+})
+export class InstallmentComponent implements OnInit {
+  InstallmentList : Installment[] =[];
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/app/pages/installments/installment/installment.component.ts:12]: property must be camel case.

File: retail-managment-system/rms-ui/src/app/pages/installments/installment/installment.component.ts [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+  }
+
+  InstallForm() {
+    this.installForm = this.formBuilder.group({
+      id: [''],
+      installmentAmount: [''],
+      paymentAmount: [''],
+      dueDate: ['', [Validators.required]],
+      paymentDate: [''],
+      status: [''],
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/app/pages/installments/installment/installment.component.ts:54]: for now you can make installmentAmount and paymentAmount required

**Review by ZinabHashem — CHANGES_REQUESTED** [review]

**Comment by ZinabHashem** [comment]
when make action delete must open popup for user to confirm that he want to delete.
 when error happen button still disabled please Handle this case.

File: retail-managment-system/rms-ui/src/app/domain/installment/models/installment.ts (hunk 1 above)
Comment (AsmaMohame) [reply retail-managment-system/rms-ui/src/app/domain/installment/models/installment.ts]: Done

File: retail-managment-system/rms-ui/src/app/pages/installments/installment/installment.component.html (hunk 2 above)
Comment (AsmaMohame) [reply retail-managment-system/rms-ui/src/app/pages/installments/installment/installment.component.html]: if you mean section 
No , I deleted it

File: retail-managment-system/rms-ui/src/app/pages/installments/installment/installment.component.ts (hunk 3 above)
Comment (AsmaMohame) [reply retail-managment-system/rms-ui/src/app/pages/installments/installment/installment.component.ts]: Done

File: retail-managment-system/rms-ui/src/app/pages/installments/installment/installment.component.ts (hunk 4 above)
Comment (AsmaMohame) [reply retail-managment-system/rms-ui/src/app/pages/installments/installment/installment.component.ts:54]: Done

File: retail-managment-system/rms-ui/src/app/pages/installments/installment/installment.component.ts [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+  openConfirmationDialog(install : Installment) {
+    let dialogRef = this.dialog.open(ConfirmDialogComponent);
+    dialogRef.afterClosed().subscribe((result) => {
+      if (result === 'yes') {
+        this.DeleteInstallment(install);
+      }
+    });
+  }
+
+  DeleteInstallment(install :Installment) {
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/app/pages/installments/installment/installment.component.ts:117]: function name must be camel case

File: retail-managment-system/rms-ui/src/app/pages/installments/installment/installment.component.ts [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+  }
+
+  getAllInstallments(): void {
+    this.installmentrepositry.getList().subscribe((result) => {
+      this.installmentList = result;
+    });
+  }
+
+
+  InstallForm() {
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/app/pages/installments/installment/installment.component.ts:49]: function name must be camel case

File: retail-managment-system/rms-ui/src/app/core/layout/side-bar/side-bar.component.html [hunk 7]
```diff
@@ -43,9 +38,4 @@
       >Products</a
     >
   </p>
-  <p>
-    <a mat-button routerLink="profits" routerLinkActive="active" class="fs-5"
-      >Profits</a
-    >
-  </p>
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/app/core/layout/side-bar/side-bar.component.html:50]: why deleted profits?

and why deleted every thing related with profits?

**Review by ZinabHashem — CHANGES_REQUESTED** [review]

File: retail-managment-system/rms-ui/src/app/core/layout/side-bar/side-bar.component.html (hunk 7 above)
Comment (AsmaMohame) [reply retail-managment-system/rms-ui/src/app/core/layout/side-bar/side-bar.component.html]: I did not commit changes related to it , but now I did

**Pull request merged**